static/app/components/globalSdkUpdateAlert.tsx

```diff
@@ -3,161 +3,113 @@ import styled from '@emotion/styled';
-  isDismissed: boolean | null;
-};
-
-type AnalyticsOpts = {
-  organization: Organization;
-};
-
-const recordAnalyticsSeen = ({organization}: AnalyticsOpts) =>
-  trackAdvancedAnalyticsEvent('sdk_updates.seen', {organization});
-
-const recordAnalyticsSnoozed = ({organization}: AnalyticsOpts) =>
-  trackAdvancedAnalyticsEvent('sdk_updates.snoozed', {organization});
-
-const recordAnalyticsClicked = ({organization}: AnalyticsOpts) =>
-  trackAdvancedAnalyticsEvent('sdk_updates.clicked', {organization});
-
-const flattenSuggestions = (list: ProjectSdkUpdates[]) =>
+const flattenSuggestions = (list: ProjectSdkUpdates[]): SDKUpdatesSuggestion[] =>
   list.reduce<SDKUpdatesSuggestion[]>(
-    (suggestions, sdk) => [...suggestions, ...sdk.suggestions],
+    (suggestions, sdk) => suggestions.concat(sdk.suggestions),
     []
   );
+interface InnerGlobalSdkSuggestionsProps extends AlertProps {
+  sdkUpdates?: ProjectSdkUpdates[] | null;
+  selection?: PageFilters;
+}
 
-class InnerGlobalSdkSuggestions extends React.Component<Props, State> {
-  state: State = {
-    isDismissed: null,
-  };
-
-  componentDidMount() {
-    this.promptsCheck();
-    recordAnalyticsSeen({organization: this.props.organization});
-  }
+function InnerGlobalSdkUpdateAlert(
+  props: InnerGlobalSdkSuggestionsProps
+): React.ReactElement | null {
+  const api = useApi();
+  const organization = useOrganization();
 
-  async promptsCheck() {
-    const {api, organization} = this.props;
+  const [showUpdateAlert, setShowUpdateAlert] = React.useState<boolean>(false);
 
-    const prompt = await promptsCheck(api, {
+  const checkIfComponentShouldPrompt = React.useCallback(() => {
+    promptsCheck(api, {
       organizationId: organization.id,
       feature: 'sdk_updates',
+    }).then(prompt => {
+      setShowUpdateAlert(!promptIsDismissed(prompt));
     });
+  }, [api, organization]);
 
-    this.setState({
-      isDismissed: promptIsDismissed(prompt),
-    });
-  }
-
-  snoozePrompt = () => {
-    const {api, organization} = this.props;
+  const handleSnoozePrompt = React.useCallback(() => {
     promptsUpdate(api, {
       organizationId: organization.id,
       feature: 'sdk_updates',
       status: 'snoozed',
     });
 
-    this.setState({isDismissed: true});
-    recordAnalyticsSnoozed({organization: this.props.organization});
-  };
-
-  render() {
-    const {
-      api: _api,
-      selection,
-      sdkUpdates,
-      organization,
-      Wrapper,
-      ...props
-    } = this.props;
-
-    const {isDismissed} = this.state;
-
-    if (!sdkUpdates || isDismissed === null || isDismissed) {
-      return null;
-    }
-
-    // withSdkUpdates explicitly only queries My Projects. This means that when
-    // looking at any projects outside of My Projects (like All Projects), this
-    // will only show the updates relevant to the to user.
-    const projectSpecificUpdates =
-      selection?.projects.length === 0 || selection?.projects === [ALL_ACCESS_PROJECTS]
```

Comment:
I think we had a bug here - `selection?.projects === [ALL_ACCESS_PROJECTS]` is comparing arrays by reference and since [ALL_ACCESS_PROJECTS] is always a new array, it will point to a new different reference and the check will always fail. I changed the check to selection?.projects[0] === ALL_ACCESS_PROJECTS so that we compare by value

Reply:
Yeah that sounds right, the original code doesn't look like it works for all.